src/api/api.c

```diff
@@ -896,6 +905,7 @@ z_queryable_options_t z_queryable_options_default(void) {
 
 z_owned_queryable_t z_declare_queryable(z_session_t zs, z_keyexpr_t keyexpr, z_owned_closure_query_t *callback,
```

Comment:
If queryables are disabled, I would completely omit any related APIs.

In doing so, the user will have an explicit error at compilation. Otherwise, it will become harder for users (and us) to debug non-compliant behavior.

Also, you can save some extra bytes in the final binary.